homeassistant/components/tis_control/switch.py

```diff
@@ -0,0 +1,167 @@
+"""Switch platform for integration_blueprint."""
+
+from __future__ import annotations
+
+from math import ceil
+from typing import Any
+
+from TISControlProtocol.BytesHelper import build_packet, int_to_8_bit_binary
+from TISControlProtocol.mock_api import TISApi
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL, STATE_OFF, STATE_ON, Platform
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_devices: AddEntitiesCallback
+) -> None:
+    """Set up the TIS switches."""
+    tis_api: TISApi = hass.data[DOMAIN]["tis_api"]
+    # Fetch all switches from the TIS API
+    switches: dict = await tis_api.get_entities(platform=Platform.SWITCH)
+    if switches:
+        # Prepare a list of tuples containing necessary switch details
+        switch_entities = [
+            (
+                appliance_name,
+                next(iter(appliance["channels"][0].values())),
+                appliance["device_id"],
+                appliance["is_protected"],
+                appliance["gateway"],
+            )
+            for switch in switches
+            for appliance_name, appliance in switch.items()
+        ]
+        # Create TISSwitch objects and add them to Home Assistant
+        tis_switches = [
+            TISSwitch(tis_api, switch_name, channel_number, device_id, gateway)
+            for switch_name, channel_number, device_id, is_protected, gateway in switch_entities
+        ]
```

Comment:
Why do we first create a tuple to then iterate again to create the entities

Reply:
it was better for debugging and the code is more readable that way